### PR TITLE
fix: Added github and gitlab token into env

### PR DESCRIPTION
**Pull request by rishabh625**
Signed-off-by: rishabh625 <rishabhmishra625@gmail.com>

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)). 

Copied tokens into env variable needed by some of tests of application set
Related #8864

### CODE REVIEW

File: .github/workflows/ci-build.yaml [hunk 1]
```diff
@@ -71,6 +71,9 @@ jobs:
     runs-on: ubuntu-latest
     needs:
       - build-go
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (alexmt) [inline .github/workflows/ci-build.yaml:75]: GITHUB_TOKEN is a reserved secret name in Github. Please use E2E_TEST_GITHUB_TOKEN and E2E_TEST_GITLAB_TOKEN instead.

**Review by alexmt — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-build.yaml (hunk 1 above)
Comment (rishabh625) [reply .github/workflows/ci-build.yaml]: thnx! done! , i hope u meant for secrets name and not env variable

**Review by jannfis — APPROVED** [review]
LGTM

The generators that depend on secrets can't be tested on a PR, maybe we should follow up on that.

File: .github/workflows/ci-build.yaml (hunk 1 above)
Comment (rishabh625) [reply .github/workflows/ci-build.yaml]: @alexmt : please re review! eagerly waiting to see this fix, post merge test runs

**Review by alexmt — APPROVED** [review]

**Pull request merged**